### PR TITLE
Enhancement/141 use node sass

### LINKED ISSUE #141: Switch to node-Sass / libsass
I think it would be good to remove the dependency of ruby-sass and use node-sass?  @himdel @chalettu @AllenBW thoughts on this?

### PR DESCRIPTION
**Pull request by AllenBW**
closes #141 

Replaces ruby dependency (gulp-ruby-sass) with gulp-sass.

No visual changes result, build process is identical.

@chriskacerguis @dtaylor113 @jeff-phillips-18 if you're curious...

Not sure what to do about the hakiri error, ruby is no longer in the app so... is it necessary?

### CODE REVIEW

**Comment by chriskacerguis** [comment]
@himdel @martinpovolny @AparnaKarve @chalettu thoughts?

/cc @Fryguy @dclarizio @chessbyte

File: gulp/config.js [hunk 1]
```diff
@@ -232,20 +232,7 @@ module.exports = (function() {
...
-        bower + 'patternfly-sass/assets/stylesheets/',
-        // Hack so that font-awesome/variables can be found
-        client + 'assets/sass/'
-      ],
-      style: 'compact',
-      noCache: false,
-      compass: false,
-      bundleExec: true,
-      sourcemap: false,
+      outputStyle: 'compact',
```
Comment (chriskacerguis) [inline gulp/config.js]: do we want to use "compress" to minify the css?

**Comment by chriskacerguis** [comment]
@Fryguy as this is no longer a "Ruby" app, thoughts about disabling "Hariri"

File: gulp/tasks/sass.js [hunk 2]
```diff
@@ -17,9 +17,8 @@ module.exports = function(gulp, options) {
       browserSync.notify('Compiling Sass');
     }
 
-    return sass(config.src, config.options)
-      .pipe(plumber({errorHandler: options.onError}))
-      .on('error', options.onError)
+    return gulp.src(config.src)
+      .pipe(sass(config.options).on('error', sass.logError))
```
Comment (himdel) [inline gulp/tasks/sass.js]: This changes how error handling works - previously, `errorHandler` from gulpfile.js would get called, respecting `--fail`, etc. Sass logError does something else, right? Might make sense to keep using `options.onError`?

Comment (AllenBW) [inline gulp/tasks/sass.js]: can do 👍

**Comment by AllenBW** [comment]
@chriskacerguis compress is the same as it was before, no issue changing though

**Comment by chriskacerguis** [comment]
I think we should start to compress, also on that note are we minifying the JS?

@martinpovolny @himdel @chalettu thoughts?

File: gulp/config.js (hunk 1 above)
Comment (himdel) [inline gulp/config.js]: Agreed on `compressed` - although the difference is negligible (before 412 KB, compact 504 KB, compressed 500 KB)

**Comment by chriskacerguis** [comment]
FYI - Going to leave this open for comments until 9am Oct 4, 2016 CST.

**Comment by himdel** [comment]
As for minifying JS, I think we should be able to do that without problems, but we're currently not - but maybe a good idea for a separate PR :) (Although, whoever is doing that, please make it compatible with babel :))

**Comment by chriskacerguis** [comment]
@himdel Minify as separate PR = totally 👍

**Comment by himdel** [comment]
@AllenBW `compress` != `compressed` ;) (https://github.com/sass/node-sass#outputstyle)

**Comment by AparnaKarve** [comment]
Wouldn't minifying JS make it harder to debug from the browser?

**Comment by chriskacerguis** [comment]
@AparnaKarve it would in a production environment, but we don't want to do that in a dev one.

**Comment by AparnaKarve** [comment]
@chriskacerguis The production environment already has the JS minified (at least that's what I've seen on a QE appliance)

**Comment by AllenBW** [comment]
@AparnaKarve @himdel @chriskacerguis true story, we already min and uglify built js , one of the changes that happened on a commit when updating the dependencies and after the eslint prs made it in(as this enabling this was being held back by a mix of outta date dependencies and absent di for some components)

**Pull request merged**